### PR TITLE
amrex: add sundials variant

### PR DESCRIPTION
**Pull request by balos1**
amrex@develop now supports enabling SUNDIALS in CMake

### CODE REVIEW

**Comment by balos1** [comment]
CC: @mic84 @balay

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -44,9 +44,12 @@ class Amrex(CMakePackage):
     variant('build_type', default='Release',
             description='The build type to build',
             values=('Debug', 'Release'))
+    variant('sundials', default=False,
+            description='Build AMReX with SUNDIALS support')
 
     # Build dependencies
     depends_on('mpi', when='+mpi')
+    depends_on('sundials@3.2.1: +ARKODE +CVODE', when='@develop +sundials')
```
Comment (balay) [inline var/spack/repos/builtin/packages/amrex/package.py]: @balos1, Is sundials@4.0.1 not needed? This was the version I was attempting to test last week - based on what was discussed in the meeting.

Comment (balos1) [reply var/spack/repos/builtin/packages/amrex/package.py]: @balay AMReX itself builds fine with SUNDIALS 3.2.1. However, I am not sure if the applications that were discussed need 4.x or not.

Comment (balos1) [reply var/spack/repos/builtin/packages/amrex/package.py]: @balay Actually, after further investigating I think I need to explore the AMReX support of SUNDIALS a little more. It builds, but I am not sure if it is actually functional.

**Comment by balos1** [comment]
Actually, after further investigating I think I need to explore the AMReX support of SUNDIALS a little more.

**Pull request closed without merging**